test/e2e/secrets_test.go

```diff
@@ -181,9 +182,36 @@ func testPrivateKeyChange(t *testing.T) {
 	err = tc.validateUpgradeableCondition(meta.ConditionTrue)
 	require.NoError(t, err, "operator Upgradeable condition not in proper state")
 
-	// Re-create the known private key so SSH connection can be re-established
-	// TODO: Remove dependency on this secret by rotating keys as part of https://issues.redhat.com/browse/WINC-655
-	require.NoError(t, tc.createPrivateKeySecret(true), "error confirming known private key secret exists")
+	// revert key changes so the test suite is able to SSH into the VM
+	require.NoError(t, tc.revertPrivateKeyChange())
+}
+
+// revertPrivateKeyChange changes the cloud private key to the known key if necessary, and waits for Machine nodes
+// to make use of it
+func (tc *testContext) revertPrivateKeyChange() error {
+	currentKey, err := tc.getCloudPrivateKey()
+	if err != nil {
+		return fmt.Errorf("error getting cloud private key: %w", err)
+	}
+	knownKey, err := os.ReadFile(gc.privateKeyPath)
+	if err != nil {
+		return fmt.Errorf("unable to read private key data from file %s: %w", gc.privateKeyPath, err)
+	}
+	if bytes.Equal(currentKey, knownKey) {
+		return nil
+	}
+	if err := tc.loadExistingNodes(); err != nil {
```

Comment:
Could you comment why this is necessary? I can't figure out why the nodes may be stale here

Reply:
you're right, it doesn't seem necessary. removed